plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -961,7 +926,7 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(
                 .map(dataFileInfoCodec::fromJson)
                 .collect(toImmutableList());
 
-        Table table = buildTable(session, schemaTableName(schemaName, tableName), location, getExternalPath(new HdfsContext(session), location), handle.isExternal());
```

Comment:
This seems to be redundant check, as individual files (and implicitly directories) are created at this point. If location would be file it should fail during directory creation earlier.